### PR TITLE
Issue #299: Add SuppressionPatchXpathFilter introductions and descriptions in the README

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #299: Add SuppressionPatchXpathFilter introductions and descriptions in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,53 @@
...
+with patch file config/defaultContext.patch, whose strategy is `context`,
+support context strategy check `MethodLength` and never suppress checks `EmptyBlock` and `HiddenField`:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+    <property name="file" value="${tp}/defaultContext.patch" />
+    <property name="strategy" value="context" />
+    <property name="supportContextStrategyChecks" value="MethodLength," />
+    <property name="neverSuppressedChecks" value="EmptyBlock,HiddenField" />
+</module>
+```
```
Comment (rdiachenko) [inline README.md:49]: Please add example for the `patchedline` strategy

File: README.md [hunk 2]
```diff
@@ -1,6 +1,53 @@
...
+Filter SuppressionPatchXpathFilter(TreeWalker level) has three different strategies that control suppression
+scope. if property `strategy` is set to `newline`, then it only accepts TreeWalker audit events for TreeWalker 
+Check violations whose line number belong to added lines in patch file and will suppress all TreeWalker Checks’ 
+violation messages which are not from added lines. if property `strategy` is set to `patchedline`, it will accept 
+all violations whose line number belong to added/changed lines in patch file. if property `strategy` is set to `context`, 
+for checks listed in `supportContextStrategyChecks`, it will not only accept violations whose line number belong to added/changed/deleted lines,
+but also consider a wider context that new code introduces violations outside of added/changed lines, but its child nodes in added/changed lines,
+for checks not listed in `supportContextStrategyChecks`, it will accept violations whose line number belong to added/changed/deleted lines in patch file.
+If there is no configured patch file or the optional is set to true and patch file was not found the Filter suppresses all audit events.
+
```
Comment (rdiachenko) [inline README.md:17]: Please make a note if it is possible to use all three properties (`checkNameForContextStrategyByTokenOrParentSet`, `supportContextStrategyChecks`, `neverSuppressedChecks`) and what's the effect.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,53 @@
...
+ | supportContextStrategyChecks                     | String has user defined Checks that support context strategy | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+ | neverSuppressedChecks                            | String has user defined TreeWalker Checks to never suppress if files are touched, split by comma. This property is useful for Checks that place violation on whole file or on not all (first/last) occurrence of cause/violated code. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+ | checkNamesForContextStrategyByTokenOrAncestorSet | String has user defined TreeWalker Checks that need modify violation nodes to their ancestor nodes to expand the context scope, split by comma | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
```
[the remark this answers is not in the document]
Comment (HuGanghui) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -1,6 +1,53 @@
...
+    <property name="file" value="config/defaultContext.patch" />
+    <property name="strategy" value="newline" />
+</module>
+```
+the following configuration fragment direts the Checker to use a SuppressionPatchXpathFilter 
+with patch file config/defaultContext.patch, whose strategy is `context`,
+support context strategy check `MethodLength` and never suppress checks `EmptyBlock` and `HiddenField`:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+    <property name="file" value="${tp}/defaultContext.patch" />
```
[the remark this answers is not in the document]
Comment (HuGanghui) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -1,6 +1,53 @@
...
+ | neverSuppressedChecks                            | String has user defined TreeWalker Checks to never suppress if files are touched, split by comma. This property is useful for Checks that place violation on whole file or on not all (first/last) occurrence of cause/violated code. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+ | checkNamesForContextStrategyByTokenOrAncestorSet | String has user defined TreeWalker Checks that need modify violation nodes to their ancestor nodes to expand the context scope, split by comma | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+
+#### Examples
+
+For example, the following configuration fragment directs the Checker to use a SuppressionPatchXpathFilter 
+with patch file config/defaultContext.patch and strategy is `newline`:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+    <property name="file" value="config/defaultContext.patch" />
```
[the remark this answers is not in the document]
Comment (HuGanghui) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (HuGanghui) [reply README.md:66]: done

File: README.md (hunk 2 above)
Comment (HuGanghui) [reply README.md:17]: done

**Review by rdiachenko — APPROVED** [review]
LGTM

**Pull request merged**